ShadowDOM/explainer.md

```diff
@@ -168,226 +168,341 @@ When asked about pain points in [Web Components](https://2023.stateofhtml.com/en
 
-The JavaScript version of this could also support CSS modules: 
+The JavaScript version of this could also support CSS modules:
 ```css
-@sheet sheet1 { 
-  :host { 
-    display: block; 
-    background: red; 
-  } 
-} 
- 
-@sheet sheet2 { 
-  p { 
-    color: blue; 
-  } 
-} 
+@sheet sheet1 {
+  :host {
+    display: block;
+    background: red;
+  }
+}
+
+@sheet sheet2 {
+  p {
+    color: blue;
+  }
+}
 ```
 ```html
-<script> 
-import {sheet1, sheet2} from './styles1and2.css' assert {type: 'css'}; 
-... 
-shadow.adoptedStyleSheets = [sheet1, sheet2]; 
+<script>
+import {sheet1, sheet2} from './styles1and2.css' assert {type: 'css'};
+...
+shadow.adoptedStyleSheets = [sheet1, sheet2];
 </script>
 ```
-This approach could be combined with other approaches listed in this document. Note that `@sheet` is not implemented by any rendering engine as of September 2024. 
+This approach could be combined with other approaches listed in this document.
+
+The specification of `@sheet` could be modified to split the *definition* of stylesheets from the *application* of the style rules. With this modification, `@sheet` would *define* a stylesheet with its own set of rules, but not  *apply* the rules automatically. This would allow for defining stylesheets in a light DOM context and applying them only to the shadow roots.
 
-### [Id-based `adoptedstylesheet` attribute on template](https://github.com/WICG/webcomponents/issues/939#issue-971914425) 
-This proposal will add a new markup-based `adoptedstylesheets` property that closely matches the existing JavaScript property.  The behavior would be just like the `adoptedStyleSheet` property that already exists in JavaScript, except it would accept a list of id attributes instead of a `ConstructableStylesheet` JavaScript object. 
+With this behavior, the following example would have a gray background and blue text only within the Shadow DOM:
 ```html
-<style type="css" id="shared_shadow_styles"> 
-    :host {  
-      color: red  
-    }  
+<style>
+  @sheet sheet1 { *: background-color: gray; }
+  @sheet sheet2 { *: color: blue; }
+</style>
+<span>I am in the light DOM</span>
+<template shadowrootmode="open" adoptedstylesheets="sheet1, sheet2">
+  <span>I'm in the shadow DOM</span>
+</template>
+ ```
+
+The light DOM could opt into particular stylesheets defined by `@sheet` via existing mechanisms such as `@import`:
+
+```html
+<style>
+  @sheet sheet1 { *: background-color: gray; }
+  @sheet sheet2 { *: color: blue; }
+  @import sheet("sheet1");
+  @import sheet("sheet2");
+</style>
+ ```
+
+A similar mechanism for `@sheet` was proposed in [this](https://github.com/w3c/csswg-drafts/issues/5629#issuecomment-2016582527) comment.
+
+Stylesheets can be adopted though multiple layers of Shadow DOM as shown in the following example:
```

Comment:
I wonder if we should add a note that because the sheets are not global, the identifier would need to be passed down at each layer if desired to share in a nested Shadow DOM context, as illustrated in the example below

Reply:
Good idea. I modified your test slightly to include the `@import` bit as well: https://github.com/MicrosoftEdge/MSEdgeExplainers/pull/905/commits/ffdcb3b80732267e53985278fd1a3200278d102b